### PR TITLE
Remove tooltip and popover styling as it is handled by Popper

### LINKED ISSUE #35039: Bootstrap 5 Sass-variable $tooltip-margin not clear or not working
When setting the V5 Sass-variable $tooltip-margin to anything but it's default 0 there seems to be no discernable effect on actual tooltips. Forcing the value by adding !important results in an offset in both x and y directions, which would be of little practical use (though noted that !important could have overriden other definitions down the line in this case).

If this behavior was replaced by the JS offset option, which is how I interpreted it, this variable should probably be removed. But feel free to close / remove this issue if this is simply lack of understanding of it's intended use.

See [this Pen](https://codepen.io/PatrickKennisnet/pen/KKqBPZB) for simulated use of the variable.

Confirmed in Chrome, Edge, Firefox on Windows 10.

### PR DESCRIPTION
**Pull request by GeoSot**
Closes #35039

Popper.js [suggests to avoid the usage of margins](https://popper.js.org/docs/v2/migration-guide/#4-remove-all-css-margins), so our variable, and some initializations are obsolete

Preview https://deploy-preview-34627--twbs-bootstrap.netlify.app/

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -1272,7 +1272,6 @@ $tooltip-border-radius:             $border-radius !default;
 $tooltip-opacity:                   .9 !default;
 $tooltip-padding-y:                 $spacer * .25 !default;
 $tooltip-padding-x:                 $spacer * .5 !default;
-$tooltip-margin:                    0 !default;
```
Comment (mdo) [inline scss/_variables.scss:1275]: This would need to get nullified and deprecated, not removed outright.

Comment (GeoSot) [reply scss/_variables.scss:1275]: @mdo I have push a commit, with the nullified variable and the deprecation message. 

TBH I am not sure, I 've done it the right way, so feel free to change what ever you like

File: scss/_tooltip.scss [hunk 2]
```diff
@@ -1,9 +1,9 @@
 // Base class
 .tooltip {
-  position: absolute;
   z-index: $zindex-tooltip;
   display: block;
   margin: $tooltip-margin;
+  @include deprecate("`$tooltip-margin`", "v5", "v5.x", );
```
[the remark this answers is not in the document]
Comment (mdo) [reply scss/_tooltip.scss:6]: Do you think we need the deprecation notice here? Wonder if Migration guide entry and null variable would be enough.

[the remark this answers is not in the document]
Comment (GeoSot) [reply scss/_tooltip.scss:6]: I am ok, with any decision.  and a full removal, would be great too. 
Please fill free to commit any change you like

[the remark this answers is not in the document]
Comment (mdo) [reply scss/_tooltip.scss:6]: Let's drop the deprecate notice and keep the `null` variable :). I'll push an update to the migration guide too then.

[the remark this answers is not in the document]
Comment (mdo) [reply scss/_tooltip.scss:6]: Just pushed and rebased.

File: scss/_variables.scss [hunk 3]
```diff
@@ -1320,7 +1320,7 @@ $tooltip-border-radius:             $border-radius !default;
...
+$tooltip-margin:                    null !default;
```
Comment (XhmikosR) [inline scss/_variables.scss:1323]: Shouldn't we add a comment to remember to drop this on v6?

**Comment by ffoodd** [comment]
I noticed a case [in RTL cheatsheet](https://deploy-preview-34627--twbs-bootstrap.netlify.app/docs/5.1/examples/cheatsheet-rtl/#tooltips) where the arrow is on the wrong side on small screen.

Not quite sure what's happening but needs some diagnostics. As far as I remember, RTL needed the `left: 0` too—but that's kinda old now in my mind 😅

**Comment by mdo** [comment]
@GeoSot If this is still good to go, want to take a crack at resolving conflicts and we can review once more before merging?

**Comment by GeoSot** [comment]
For v5 seems fine. As popper has been changed lately, I am not aware if theses changes will be reverted on BS v6

**Review by mdo — APPROVED** [review]

**Pull request merged**